### DISCUSSION TITLE
Cluster of uWeboscket.js

### DISCUSSION THREAD

**Discussion by emdotem** [uNetworking/uWebSockets.js#214] [Q&A] [answered]
Is it possible to have multiple instances of the server and make sure the communication among sockets connected to different instance works?

I'm thinking about having a message broker in the middle for communication

**Comment by dioptre** [comment]
i use nats for that, works well

**Comment by ghost** [comment, marked as answer]
You can scale with multiple instances, yes. Either listen to the same port (Linux only) or listen to a set of ports which are then proxied or whatever. You can also scale using DNS, etc.

**Comment by dalisoft** [comment]
@emdotem It is works only in Linux, just add `NODE_CLUSTER_SCHED_POLICY=rr` as ENV to be sure everything will work as expected. And works so fast, i checked

**Comment by ghost** [comment]
NODE_CLUSTER_SCHED_POLICY does not affect uws

**Comment by yovanoc** [comment]
Sorry to comment this closed issue, I hope you will see it but when you talk about cluster you mean the nodejs cluster module https://nodejs.org/docs/latest/api/cluster.html ?

**Comment by ghost** [comment]
It doesn't matter. I'm pretty sure Workers works also. Or just start separate processes altogether. None of it matters, it is the Linux kernel which does the load balancing. Or use a proxy it doesn't matter

**Reply by kuabhish** [reply]
hello
if i start a separate peocess how would we be able to communicate between connections in different process.

if we use a proxy then event loop of the proxy will wait i think so latency wont improve.. please tell me if m wrong

**Reply by ezioda004** [reply]
@kuabhish By using a message broker like redis, rabbitmq, or if you're running multiple uws process in the same host using node cluster then you can do IPC.

**Reply by johnsonfash** [reply]
> @kuabhish By using a message broker like redis, rabbitmq, or if you're running multiple uws process in the same host using node cluster then you can do IPC.

How do i communicate with different process using IPC. the way i do it is using ws with Map() and thats when you have one instance running. 

With works, i belive you have one instance but the others ie load are running with all the cores but they share same truth so 

`const  users = new Map<UWSConnection>()`

might work, but how to go about this is unknown
